doc/helpers/responsive-extension.md

```diff
@@ -0,0 +1,127 @@
+---
+uid: Toolkit.UI.ResponsiveExtension
+---
+# ResponsiveExtension
+
+## Summary
+The `ResponsiveExtension` class is a markup extension in the Uno Toolkit UI library that allows you to adapt the properties of a UI element based on the current window width.
+This is particularly useful for creating responsive designs that look good on a variety of screen sizes.
+
+### C#
+```csharp
+public partial class ResponsiveExtension : MarkupExtension, IResponsiveCallback
+```
+
+### XAML
+```xml
+xmlns:utu="using:Uno.Toolkit.UI"
+...
+
+<utu:ResponsiveExtension />
+```
+
+### Inheritance 
+Object &#8594; MarkupExtension &#8594; ResponsiveExtension
+
+### Constructors
+| Constructor           | Description                                                    |
+|-----------------------|----------------------------------------------------------------|
+| ResponsiveExtension() | Initializes a new instance of the `ResponsiveExtension` class. |
+
+### Properties
+| Property   | Type             | Description                                                                                         |
+| ---------- | ---------------- | --------------------------------------------------------------------------------------------------- |
+| Narrowest  | object           | Value to be used when the screen size is at its narrowest.                                          |
+| Narrow     | object           | Value to be used when the screen size is narrow.                                                    |
+| Normal     | object           | Value to be used when the screen size is normal.                                                    |
+| Wide       | object           | Value to be used when the screen size is wide.                                                      |
+| Widest     | object           | Value to be used when the screen size is at its widest.                                             |
+| Layout     | ResponsiveLayout | Overrides the MaxWidth for each screen size: `Narrowest`, `Narrow`, `Normal`, `Wide`, and `Widest`. |
+
+### Usage
+```xml
+<!-- Include the following XAML namespace to use the sample below -->
+xmlns:utu="using:Uno.Toolkit.UI"
+...
+<Page.Resources>
+	<SolidColorBrush x:Key="NarrowRed">Crimson</SolidColorBrush>
+	<SolidColorBrush x:Key="WideBlue">Blue</SolidColorBrush>
+</Page.Resources>
+...
+<Border Width="30"
+		Height="30"
+		HorizontalAlignment="Left"
+		Background="{utu:Responsive Normal={StaticResource NarrowRed},
+									Wide={StaticResource WideBlue}}" />
+```
+
+## Remarks
+**Platform Limitations**: The ability to update property values when the window size changes is only available on targets other than Windows UWP. On UWP targets, where only the parameterless constructor is available, the property value can only be set initially and cannot be updated if the window size changes. This is a limitation of UWP, and the class logs a warning message when the extension is used on UWP.
```

Comment:
```suggestion
**Platform Limitations**: The ability to update property values when the window size changes is only available on targets other than Windows UWP. Due to limitation of UWP api (only for windows target), the `MarkupExtension.ProvideValue(IXamlServiceProvider)` overload is unavailable, which is required to continuously update the value. Because of this, this markup-extension will only provide the initial value, and will not respond to window size change.
```

Reply:
updated